crates/bevy_ecs/src/change_detection.rs

```diff
@@ -173,6 +173,17 @@ macro_rules! impl_into_inner {
                 self.set_changed();
                 self.value
             }
+
+            /// Maps to an inner value by applying a function to the contained reference, without flagging a change.
+            ///
+            /// You must *not* modify the argument passed to the closure.
```

Comment:
Nit: The word "must" sounds a bit strong for something that doesn't cause UB. Maybe replace it with "should"?
```suggestion
            /// You should *not* modify the argument passed to the closure.
```
Or keep it if you feel it needs to be made a strong warning.